### DISCUSSION TITLE
Copy city names

### DISCUSSION THREAD

**Discussion by Leeb64** [devemux86/cruiser#280] [Q&A] [answered]
Would it be possible to add a function to copy the name of a place selected on the map to paste it into an APP like Benzipreis-Blitz for example?

**Comment by devemux86** [comment]
This can be possible using the "Show address" or searching offline POI.

We could add the "Name" as option in the "Share" menu?
Currently we can copy / share map links, coordinates, geo, etc.

**Reply by Leeb64** [reply]
Displaying addresses works, but copying the city name does not work. Would be handy in foreign countries where they have such weird names that are hard to remember...

**Reply by ramondevic** [reply]
Yes, showing "Show address" you can see the name of the place, street, town, ... perhaps all this could also be shared on the clipboard so you can also take it from a text editor to use it elsewhere.

**Reply by devemux86** [reply]
I can add "Address" to the "Share" menu as another option.

**Comment by ramondevic** [comment]
I don't know if "Address" or better an option with the information that is shown now when doing "Show address" more coordinates and that it is saved to the clipboard in text format.

![Sense títol](https://github.com/devemux86/cruiser/assets/122691401/2e740c61-b2b5-4c77-8f5a-0d18317e4ac4)

**Reply by devemux86** [reply]
The other options in the image (plus code, what3words, coordinates) are already available in the "Share" menu.
(with automatic copy to clipboard)

Copying everything together could be confusing, as most may only want one data of all.

**Reply by ramondevic** [reply]
Sometimes I need to give an address to someone who is not a user of Cruiser or other navigation tools and what they need is the place name, address and town. I give it to him in TEXT format because it is more comfortable for him
If there are also the coordinates it can always go well.
Many people I know are not interested in "plus code" or "what3words"

To avoid having huge options menus, maybe you could add the coordinates to the same box that I put before.
And it would be great if you could create a text in the clipboard, as it is in the box. My receptor will forget "plus code" and "what3words", no problem.

IF it can't be done, I'll write the data on WhatsApp as I'm doing now :(

Thanks.

**Reply by devemux86** [reply]
Do you mean copy address + coordinates or copy everything?

Plus codes and what3words were designed to work independently of coordinates or addresses.

I'm trying to understand the use case.

**Reply by ramondevic** [reply]
I mean copy everything, to be able to paste in an email or a WhatsApp message, in text mode so that a person can read and understand it and not to send it automatically to any application.

You can use exactly the same information from the "show address" window.

**Reply by Leeb64** [reply]
It would be enough for me if I could copy the city name and postal code from the "Show address" field to paste it into the "Benzinpreis Blitz" app.

**Reply by devemux86** [reply]
I can add an "Address" option in "Share" with the nearest address data (then you can keep what you need).

And probably something like "All" with more data, like address, coords, codes, etc.

**Reply by devemux86** [reply]
@Leeb64 doesn't this app work with more precise data, like coordinates or geo: ?

**Reply by Leeb64** [reply]
Unfortunately, these and other of these petrol price APPs only work with the city name or the postcode

**Comment by devemux86** [comment]
@Leeb64 is this app?

https://play.google.com/store/apps/details?id=de.mwwebwork.benzinpreisblitz

**Reply by devemux86** [reply]
I tested that app.

Next Cruiser will have an "Address" option in the "Share" menu copying the nearest address.
Map locations may return different details depending on area geocoding: address, zip, city, ...

**Reply by devemux86** [reply]
> Next Cruiser will have an "Address" option in the "Share" menu copying the nearest address.

@Leeb64 you can test it in:

- https://github.com/devemux86/cruiser/discussions/299

**Reply by Leeb64** [reply]
Einwandfrei!

**Comment by devemux86** [comment, marked as answer]
> Next Cruiser will have an "Address" option in the "Share" menu copying the nearest address.

Available in [Cruiser 3.0.32](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
